ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13130,8 +13131,7 @@ public void validate() throws SemanticException {
 
           LOG.debug("validated " + usedp.getName());
           LOG.debug(usedp.getTable().getTableName());
-          WriteEntity.WriteType writeType = writeEntity.getWriteType();
-          if (writeType != WriteType.UPDATE && writeType != WriteType.DELETE) {
+          if (!transactionalInQuery && conf.getBoolVar(HIVEARCHIVEENABLED)) {
```

Comment:
@pvary 
the variable is set based on that only above:
```
          if (AcidUtils.isTransactionalTable(tbl)) {
            transactionalInQuery = true;
```
At L13124, rather than re-doing AcidUtils.isTransactionalTable(tbl), making use of the variable

Reply:
What happens with `transactionalInQuery` when we have multiple tables in the query? Say 1 transactional, and 1 with archives?